src/main/scala/pipedsl/common/Syntax.scala

```diff
@@ -228,16 +422,29 @@ object Syntax {
   case class ECall(mod: Id, args: List[Expr]) extends Expr
   case class EVar(id: Id) extends Expr
   case class ECast(ctyp: Type, exp: Expr) extends Expr
+  {
+    typ = Some(ctyp)
+  }
 
 
-  sealed trait Command extends Positional with SMTPredicate with PortAnnotation
+  sealed trait Command extends Positional with SMTPredicate with PortAnnotation with HasCopyMeta
+  {
+    override val copyMeta: HasCopyMeta => Command =
+      {
+        case from :Command =>
+        setPos(from.pos)
+        portNum = from.portNum
+        predicateCtx = from.predicateCtx
+        this
+      }
+  }
   case class CSeq(c1: Command, c2: Command) extends Command
   case class CTBar(c1: Command, c2: Command) extends Command
   case class CIf(cond: Expr, cons: Command, alt: Command) extends Command
-  case class CAssign(lhs: EVar, rhs: Expr) extends Command{
+  case class CAssign(lhs: EVar, rhs: Expr, typ: Option[Type]) extends Command{
```

Comment:
Is there a reason we need a new field for this and can't use the existing `typ` variable?